crates/uv-distribution-filename/src/wheel.rs

```diff
@@ -319,7 +319,15 @@ impl TryFrom<&Url> for WheelFilename {
                     "URL must contain a filename".to_string(),
                 )
             })?;
-        Self::from_str(filename)
+
+        // Strip query parameters from the filename, if they exist.
+        let filename_without_query = if let Some(query_start) = filename.rfind('?') {
+            &filename[..query_start]
+        } else {
+            filename
+        };
```

Comment:
I think we also want to strip fragments (`#` at the end). There's an implementation for this in `URLString`:

```rust
    /// Return the [`UrlString`] with any query parameters and fragments removed.
    pub fn base_str(&self) -> &str {
        self.as_ref()
            .split_once('?')
            .or_else(|| self.as_ref().split_once('#'))
            .map(|(path, _)| path)
            .unwrap_or(self.as_ref())
    }
```

Reply:
If you prefer, I can make this patch and merge.

Reply:
Yes, please!

Reply:
It looks like this code is actually unused (https://github.com/astral-sh/uv/pull/14223) so this isn't quite the right spot. I commented back in the originating issue.